pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+package nsmconnect;
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string name = 1;
+    string namespace = 2;
+    Label labels = 3;
+}
+
+message KernelInterface {
+   Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message VhostUserInterface{
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message MemInterface {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+enum InterfaceType {
+    kernel_interface = 0;
+    vhost_user = 1;
+    memif = 2;
+  }
+
+enum InterfacePreference {
+    first = 0;
+    second = 1;
+    third = 2;
+    forth = 3;
+    fifth = 4;
+}
+
+message MechanismType {
+    InterfaceType type = 1;
+    InterfacePreference preference = 2;
+  }
+
+message AffinityPreference {
+    string predicate = 1;
+}
+
+message ConnectionRequest {
+
+    Metadata metadata = 1;
+    
+    string network_service_name = 2;
+
+    repeated MechanismType type = 3;
+
+    repeated AffinityPreference predicate = 4;
+
+    string device_claim = 5;
+
+    KernelInterface kernel_interface = 6;
+
+    VhostUserInterface vhost_user_interface = 7;
+
+    MemInterface mem_interface = 8;
+}
+
+message ConnectionParameters {
+    string address = 1;
+    repeated string route = 2;
+}
+
+message ConnectionAccept {
+    bool accepted = 1;
+
+    string admission_error = 2;
+
+    repeated MechanismType accepted_type = 3;
+
+    KernelInterface kernel_interface = 4;
+
+    VhostUserInterface vhost_user_interface = 5;
+
+    MemInterface mem_interface = 6;
+}
+
+// Copy of netmesh.proto since I could not yet figure out a correct way to import it 
```

Comment:
You do something like this:

```
import "netmesh/model/netmesh/netmesh.proto";
```

Reply:
I tried exactly the same and go generate fails :( will give it another shot.

Reply:
Is it version dependent perhaps?